src/main/kotlin/no/nav/syfo/service/DialogmoteInnkallingVarselService.kt

```diff
@@ -0,0 +1,144 @@
+        log.info("[DIALOGMOTE_STATUS_VARSEL_SERVICE]: sender dialogmote hendelse til narmeste leder ${varselHendelse.type}")
+        sendVarselTilDineSykmeldte(varselHendelse)
+        sendVarselTilArbeidsgiverNotifikasjon(varselHendelse)
+    }
+
+    fun sendVarselTilArbeidstaker(varselHendelse: ArbeidstakerHendelse) {
+        val url = URL(dialogmoterUrl + BRUKERNOTIFIKASJONER_DIALOGMOTE_SYKMELDT_URL)
+        val text = getArbeidstakerVarselText(varselHendelse.type)
+        if (!text.isNullOrBlank()) {
+            senderFacade.sendTilBrukernotifikasjoner(
+                UUID.randomUUID().toString(),
+                varselHendelse.arbeidstakerFnr,
+                text,
+                url,
+                varselHendelse
+            )
+        }
+    }
+
+    private fun sendVarselTilArbeidsgiverNotifikasjon(varselHendelse: NarmesteLederHendelse) {
+        val texts = getArbeisgiverTexts(varselHendelse)
+        val sms = texts[SMS_KEY]
+        val emailTitle = texts[EMAIL_TITLE_KEY]
+        val emailBody = texts[EMAIL_BODY_KEY]
+
+        if (!sms.isNullOrBlank() && !emailTitle.isNullOrBlank() && !emailBody.isNullOrBlank()) {
+            val input = ArbeidsgiverNotifikasjonInput(
+                UUID.randomUUID(),
+                varselHendelse.orgnummer,
+                varselHendelse.narmesteLederFnr,
+                varselHendelse.arbeidstakerFnr,
+                ARBEIDSGIVERNOTIFIKASJON_DIALOGMOTE_MERKELAPP,
+                sms,
+                emailTitle,
+                emailBody,
+                LocalDateTime.now().plusWeeks(WEEKS_BEFORE_DELETE)
+            )
+
+            senderFacade.sendTilArbeidsgiverNotifikasjon(
+                varselHendelse,
+                input
+            )
+        }
+    }
+
+    private fun sendVarselTilDineSykmeldte(varselHendelse: NarmesteLederHendelse) {
+        val varselText = getDineSykmeldteVarselText(varselHendelse.type)
+        if (!varselText.isNullOrBlank()) {
+            val dineSykmeldteVarsel = DineSykmeldteVarsel(
+                ansattFnr = varselHendelse.arbeidstakerFnr,
+                orgnr = varselHendelse.orgnummer,
+                oppgavetype = varselHendelse.type.toDineSykmeldteHendelseType().toString(),
+                lenke = null,
+                tekst = varselText,
+                utlopstidspunkt = OffsetDateTime.now().plusWeeks(WEEKS_BEFORE_DELETE)
+            )
+            senderFacade.sendTilDineSykmeldte(varselHendelse, dineSykmeldteVarsel)
+        }
+    }
+
+    private fun getArbeidstakerVarselText(hendelseType: HendelseType): String? {
+        return when (hendelseType) {
+            SM_DIALOGMOTE_INNKALT -> SM_DIALOGMOTE_INNKALT_TEKST
+            SM_DIALOGMOTE_AVLYST -> SM_DIALOGMOTE_AVLYST_TEKST
+            SM_DIALOGMOTE_NYTT_TID_STED -> SM_DIALOGMOTE_NYTT_TID_STED_TEKST
+            SM_DIALOGMOTE_REFERAT -> SM_DIALOGMOTE_REFERAT_TEKST
+            else -> null
```

Comment:
Her kan du heller kaste en exception i else, feks en IllegalArgumentException. Da kan returtypen være `String` i stedet for `String?`, og da slipper du null-sjekk der du bruker `getArbeidstakerVarselText()`